src/Swashbuckle.AspNetCore.Newtonsoft/SchemaGenerator/NewtonsoftDataContractResolver.cs

```diff
@@ -83,13 +83,14 @@ public DataContract GetDataContractForType(Type type)
                 if (keyType.IsEnum)
                 {
                     // This is a special case where we know the possible key values
-                    var enumValuesAsJson = keyType.GetEnumValues()
-                        .Cast<object>()
-                        .Select(JsonConverterFunc);
-
-                    keys = enumValuesAsJson.Any(json => json.StartsWith("\""))
-                        ? enumValuesAsJson.Select(json => json.Replace("\"", string.Empty))
-                        : keyType.GetEnumNames();
+                    keys = keyType.GetEnumValues().Cast<object>().Select(v =>
+                    {
+                        var dic = new Dictionary<object, int>() { { v, 0 } };
+                        var serialized = JsonConvert.SerializeObject(dic, _serializerSettings);
+                        var deserialized =
+                            JsonConvert.DeserializeObject<Dictionary<string, int>>(serialized, _serializerSettings);
+                        return deserialized.Keys.Single();
+                    });
```

Comment:
Suggested refactors for naming, to reduce the memory usage and consistency:

```suggestion
                    keys = keyType.GetEnumValues().Cast<object>().Select(v =>
                    {
                        var dictionary = new Dictionary<object, byte>(1) { [v] = 0 };
                        var serialized = JsonConverterFunc(dictionary);
                        var deserialized = JsonConvert.DeserializeObject<Dictionary<string, byte>>(serialized, _serializerSettings);
                        return deserialized.Keys.First();
                    });
```